kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/zookeeper/ZookeeperClientProvider.scala

```diff
@@ -38,6 +39,8 @@ import org.apache.kyuubi.util.reflect.DynConstructors
 
 object ZookeeperClientProvider extends Logging {
 
+  val jaasConfigurationCache = new ConcurrentHashMap[(String, String), Configuration]()
```

Comment:
can you add comment here to explain the benefit of caching the JaasConfiguration?